### PR TITLE
[GTK] Unconditionally enable network access from the flatpak build sandbox

### PR DESCRIPTION
**Pull request by philn**
#### 9878e30a6f7350aca54e860251f34971e6aed1f6
<pre>
[GTK] Unconditionally enable network access from the flatpak build sandbox
<a href="https://bugs.webkit.org/show_bug.cgi?id=276588">https://bugs.webkit.org/show_bug.cgi?id=276588</a>

Reviewed by Alicia Boya Garcia.

The clangd-related tools added in bug #274909 require network access for the installation of the
jinja Python package. Until now the WPE build sandbox had unconditional network access already, but
the GTK build sandbox didn&apos;t. That was conditioned by the IceCC and/or SCCache options.

* Tools/flatpak/flatpakutils.py:
(WebkitFlatpak.run_in_sandbox):

Canonical link: <a href="https://commits.webkit.org/280966@main">https://commits.webkit.org/280966@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/aa2b0437b86567e9eb948cfd807ec313a27c9559

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/58135 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/37463 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/10616 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/61760 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/8580 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/60263 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/45099 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/8772 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/47101 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/6115 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/60165 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/35123 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/50255 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/27935 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/31886 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/7552 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/7584 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/53835 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/7819 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/63461 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/2049 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/7881 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/54389 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/2056 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/50267 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/54475 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/12866 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/1751 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/33292 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/34378 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/35462 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/34123 "Built successfully") | | |

### CODE REVIEW

File: Tools/flatpak/flatpakutils.py [hunk 1]
```diff
@@ -955,15 +947,13 @@ def envvar_in_suffixes_to_keep(envvar):
...
                 self.run_in_sandbox("sccache", "--start-server", env=sccache_environment, building_local_deps=building_local_deps,
-                                    extra_flatpak_args=[share_network_option], start_sccache=False)
+                                    start_sccache=False)
 
             # Forward sccache server env vars to sccache clients.
             sandbox_environment.update(sccache_environment)
 
         if self.use_icecream and not skip_icc:
             _log.debug('Enabling the icecream compiler')
-            flatpak_command.extend([share_network_option, "--filesystem=home"])
```
Comment (ntrrgc) [inline Tools/flatpak/flatpakutils.py:965]: Why are you removing the `--filesystem=home` from here?

Comment (philn) [reply Tools/flatpak/flatpakutils.py:965]: It's useless because `--filesystem=host` is added already.

Comment (ntrrgc) [reply Tools/flatpak/flatpakutils.py:965]: I see.

**Review by ntrrgc — APPROVED** [review]
LGTM.

**Pull request merged**